addons/html_editor/static/src/main/movenode_plugin.js

```diff
@@ -405,16 +406,40 @@ export class MoveNodePlugin extends Plugin {
             const [position, focusElelement] = this._currentDropHintElementPosition;
             this._currentDropHintElementPosition = undefined;
             const previousParent = movableElement.parentElement;
+
+            const isFocusInsideList = ["UL", "OL"].includes(focusElelement?.parentElement?.tagName);
+            const isMovableLI = movableElement.tagName === "LI";
+            if (isMovableLI && !isFocusInsideList) {
+                // If LI is moved outside a list, wrap it in a UL/OL (based on original parent)
+                const wrapperList = document.createElement(previousParent.tagName);
+                if (
+                    wrapperList.tagName === "UL" &&
+                    previousParent.classList.contains("o_checklist")
+                ) {
+                    wrapperList.classList.add("o_checklist");
+                }
+                wrapperList.appendChild(movableElement);
+                movableElement = wrapperList;
```

Comment:
`o_checklist` is something related to list_plugin so it should not be directly used here. Even better you could just clone the `previousParent`.
```suggestion
                // If LI is moved outside a list, wrap it in a UL/OL (based on original parent)
                const wrapperList = previousParent.cloneNode(false);
                wrapperList.appendChild(movableElement);
                movableElement = wrapperList;
```